datashape/coretypes.py

```diff
@@ -402,6 +402,67 @@ def to_numpy_dtype(self):
         return np.dtype('O', metadata={'vlen': unicode})
 
 
+class Decimal(Unit):
+    """Decimal type corresponding to SQL Decimal/Numeric types.
+
+    The first parameter passed specifies the number of digits of precision that
+    the Decimal contains. If an additional parameter is given, it represents
+    the scale, or number of digits of precision that are after the decimal
+    point.
+
+    The Decimal type makes no requirement of how it is to be stored in memory,
+    therefore, the number of bytes needed to store a Decimal for a given
+    precision will vary based on the platform where it is used.
+
+    >>> Decimal(18)
+    Decimal(18, 0)
+    >>> Decimal(7,4)
+    Decimal(7, 4)
+    """
+
+    cls = MEASURE
+    __slots__ = 'precision', 'scale'
+
+    def __init__(self, precision, *args):
```

Comment:
No, that is better. I was going back and forth on required vs. optional arguments. `scale=0` works since it'll still accept either one or two positional arguments.